server.js

```diff
@@ -18,7 +22,89 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+  res.json(endpoints);
+});
+
+// Get all thoughts
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find()
+    .sort({ createdAt: -1 })
+    .limit(20)
+    .exec();
+  res.json(thoughts);
+});
+
+// Post a thought
+app.post("/thoughts", async (req, res) => {
+  const { message } = req.body;
+  try {
+    if (message) {
+      const newThought = await new Thought({ message }).save();
+      res.status(201).json(newThought);
+    } else {
+      throw new Error("No message input...");
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      error: error.message,
+      message: "Could not save the thought...",
+    });
+  }
+});
+
+// Like a post
+app.post("/thoughts/:thoughtId/like", async (req, res) => {
+  const { thoughtId } = req.params;
+  try {
+    const post = await Thought.findByIdAndUpdate(
+      thoughtId,
+      {
+        $inc: { hearts: 1 },
+      },
+      { new: true }
+    ).exec();
+    res.status(201).json(post);
+  } catch (error) {
+    if (error.name === "CastError") {
+      error.message = `There's no post matching id:${thoughtId}`;
+    }
+    res.status(400).json({
+      success: false,
+      error: error.message,
+      message: "Could not like the thought...",
+    });
+  }
+});
+
+//unlike a post
+app.post("/thoughts/:thoughtId/unlike", async (req, res) => {
```

Comment:
Love this! What do you think about having both the like and dislike in the same endpoint though? Maybe some kind of param differing them, since both are changing the likes?